pow/src/main/java/tech/pegasys/teku/pow/DepositFetcher.java

```diff
@@ -158,14 +158,11 @@ private synchronized void postEventsForBlock(
   private DepositsFromBlockEvent createDepositFromBlockEvent(
       final EthBlock.Block block,
       final List<DepositContract.DepositEventEventResponse> groupedDepositEventResponse) {
-    return new DepositsFromBlockEvent(
+    return DepositsFromBlockEvent.create(
         UInt64.valueOf(block.getNumber()),
         Bytes32.fromHexString(block.getHash()),
         UInt64.valueOf(block.getTimestamp()),
-        groupedDepositEventResponse.stream()
-            .map(Deposit::new)
-            .sorted(Comparator.comparing(Deposit::getMerkle_tree_index))
```

Comment:
Are we sure that, without sorting the deposits for their merkle tree index, they will be in correct order?